paper.tex

```diff
@@ -367,43 +381,44 @@ \subsection{Pipeline Design}
+    }
+    \label{table:pathtag}
+\end{table}
 
-\begin{itemize}
-    \item 0-1: coordinate count
-    \item 2: subpath end bit
-    \item 3: 16/32 bit coordinates
-    \item 4: path end bit
-    \item 5: transform bit
-    \item 6: line style bit
-\end{itemize}
-
-For most path segments, the coordinate count indicates the type of path segment: 1 for lines, 2 for quadratic Beziers, and 3 for cubic Beziers. The subpath bit is set on the last segment of a subpath, and the path bit is additionally set on the last bit of a path. The latter is not directly interpreted by the path flattening code, but is forwarded to later rendering stages, for example, the path id can be used to select a color or other paint style.
-
-The size is readily inferred from the tag byte: the number of coordinate pairs is the coordinate count plus one if the subpath end bit is set. The nuber of bytes per coordinate pair is 4 for 16 bit coordinates or 8 for 32 bit coordinates.
+Table~\ref{table:pathtag} shows the individual fields of the 8-bit path tag. The offset increments for the transform and style streams can be stored in a single bit that is set to $1$ only if a path tag marks the beginning of a new transform and/or style entry. We also encode an additional \emph{path bit} in the final segment of all paths, which allows us to access additional streams (such as a fill color or gradient parameters) using a per-path offset in later rendering stages.
 
-It is relatively straightforward to convert the standard moveto/lineto/curveto path representation into this encoding. Subpath boundaries are moved from the beginning of a subpath (moveto) to the end. A separate segment must be generated for closepath if the ending point does not coincide with the start point.
+The handling of the subpath bit allows for overlap in coordinates, so that except at subpath boundaries, the first coordinate pair of each segment overlaps with the last coordinate pair of the previous one. During encoding, subpath boundaries are moved from the beginning of a subpath (moveto) to the end. An additional line segment is inserted into the tag stream for fills (closepath) if the final point does not coincide with the start point. The final segment of every subpath has the \emph{subpath end bit} set to $1$. For strokes, the subpath end segment represents the \emph{stroke cap marker} defined in Section \ref{subsec:pipeline-design}. This special segment is assigned a coordinate count of $2$, and the corresponding two points in the path data stream encode the tangent vector of the subpath's initial segment, used to correctly draw the start cap or the connecting join in a closed contour.
```

Comment:
I think "is inserted into" is a bit too passive. Maybe "The CPU-side encoding process inserts an additional line segment into the tag and coordinate streams to close the subpath if the final point does not coincide with the start point."